### PR TITLE
Introduce scalding sub-project

### PR DESCRIPTION
**Pull request by lancelet**
- Moves everything that depends on Scalding from `au.com.cba.omnia.maestro.core` to a new `au.com.cba.omnia.maestro.scalding` subproject
- Makes both maestro.core and maestro.macros independent of Scalding

Questions:
- This introduces `au.com.cba.omnia.maestro.scalding.scalding` as a package name, which seems pretty undesirable. Any suggestions for an alternative? Should I perhaps move the things which were originally in `au.com.cba.omnia.maestro.core.scalding` to the `au.com.cba.omnia.maestro.scalding` top-level sub-project package?
- I had to add a `dependencyOverrides` to the scalding sub-project for scalacheck in order to satisfy the strict dependency settings; was that the correct thing to do?

### CODE REVIEW

**Comment by lancelet** [comment]
NB: AFAICT, the codacy issues were all pre-existing. :smile:

**Comment by stephanh** [comment]
You touched it last :smile:.

Don't worry about it. I'll try and have a look at this on Monday.

File: maestro-api/src/main/scala/au/com/cba/omnia/maestro/api/package.scala [hunk 1]
```diff
@@ -17,24 +17,24 @@ package au.com.cba.omnia.maestro
...
-  type JobFailure      = au.com.cba.omnia.maestro.core.scalding.JobFailure
-  val  JobFinished     = au.com.cba.omnia.maestro.core.scalding.JobFinished
-  val  JobNotReady     = au.com.cba.omnia.maestro.core.scalding.JobNotReady
-  val  JobNeverReady   = au.com.cba.omnia.maestro.core.scalding.JobNeverReady
-  val  JobFailure      = au.com.cba.omnia.maestro.core.scalding.JobFailure
-  val MaestroExecution = au.com.cba.omnia.maestro.core.scalding.MaestroExecution
-  val MX               = au.com.cba.omnia.maestro.core.scalding.MaestroExecution
-
-  type HiveTable[A <: ThriftStruct, B] = au.com.cba.omnia.maestro.core.hive.HiveTable[A, B]
+  type JobStatus       = au.com.cba.omnia.maestro.scalding.scalding.JobStatus
```
Comment (stephanh) [inline maestro-api/src/main/scala/au/com/cba/omnia/maestro/api/package.scala]: I'm inclined to remove the intermediary `scalding` and just have `au.com.cba.omnia.maestro.scalding.` and `au.com.cba.omnia.maestro.task.`.

Does anyone else have any thoughts on this?

**Comment by stephanh** [comment]
@lancelet looks sensible to me. I'll raise it with the datahub squad leads.

File: maestro-api/src/main/scala/au/com/cba/omnia/maestro/api/package.scala (hunk 1 above)
Comment (lancelet) [inline maestro-api/src/main/scala/au/com/cba/omnia/maestro/api/package.scala]: So you'd have these two sub-packages (`scalding` and `task`), but still have them both inside the `maestro-scalding` sub-project? Or would you want two separate sub-projects? (I'm assuming the former, but I thought I'd check.)

Comment (stephanh) [inline maestro-api/src/main/scala/au/com/cba/omnia/maestro/api/package.scala]: The first one.

**Comment by lancelet** [comment]
I've now performed the package re-shuffle as discussed.

**Comment by stephanh** [comment]
:+1:.

**Pull request merged**